File: SongJSeop/week14/1074.py
def find(N, r, c):
    if N == 0:
        return 0

    halfLineCount = 2 ** (N - 1)
    quaterCount = halfLineCount ** 2
    if r >= halfLineCount and c >= halfLineCount:
        return find(N - 1, r - halfLineCount, c - halfLineCount) + (quaterCount * 3)
    elif r >= halfLineCount > c:
        return find(N - 1, r - halfLineCount, c) + (quaterCount * 2)
    elif r < halfLineCount <= c:
        return find(N - 1, r, c - halfLineCount) + quaterCount
    else:
        return find(N - 1, r, c)

n, R, C = map(int, input().split())
print(find(n, R, C))

File: iam454/week34/1074.js
const fs = require("fs");

const input = fs
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "./input.txt")
  .toString()
  .trim();

const [N, r, c] = input.split(" ").map(Number);

function z(n, r, c) {
  if (n === 0) {
    return 0;
  }

  const half = 1 << (n - 1);
  const area = half * half;

  if (r < half && c < half) {
    return z(n - 1, r, c);
  } else if (r < half && c >= half) {
    return area + z(n - 1, r, c - half);
  } else if (r >= half && c < half) {
    return 2 * area + z(n - 1, r - half, c);
  } else {
    return 3 * area + z(n - 1, r - half, c - half);
  }
}

console.log(z(N, r, c));
